src/sagas/delete-project.saga.test.js

```diff
@@ -0,0 +1,170 @@
+        name: 'apple',
+        path: '/path/to/apple',
+      };
+
+      const projects = [
+        project,
+        { id: 'b', name: 'berry', path: '/path/to/berry' },
+      ];
+
+      const saga = deleteProject({ project });
+
+      expect(saga.next().value).toEqual(
+        call(electron.remote.dialog.showMessageBox, {
+          type: 'warning',
+          buttons: ['Delete from Disk', 'Cancel'],
+          defaultId: 0,
+          title: `Delete ${project.name}`,
+          message: `Are you sure you want to delete ${project.name}?`,
+          detail: 'WARNING! Deleting from disk will send the project to trash!',
+        })
+      );
+
+      // Next, we select the projects, passing in `0` to confirm the prompt
+      // (`0` is the ID of the "Delete from Disk" option).
+      expect(saga.next(0).value).toEqual(select(getProjectsArray));
+
+      expect(saga.next(projects).value).toEqual(
+        call(electron.remote.shell.moveItemToTrash, project.path)
+      );
+
+      expect(saga.next(true).value).toEqual(
+        put(finishDeletingProjectFromDisk(project.id))
+      );
+
+      // Because there's another project, it should select the next one.
+      expect(saga.next().value).toEqual(put(selectProject('b')));
+      expect(saga.next().done).toEqual(true);
+    });
+
+    it('prompts the user to create a new project, when deleting the final project', () => {
+      const project = {
+        id: 'a',
+        name: 'apple',
+        path: '/path/to/apple',
+      };
+
+      const projects = [project];
+
+      const saga = deleteProject({ project });
+
+      // Show dialog
+      saga.next();
+      // Confirm dialog
+      saga.next(0);
+      // Pass in the projects to the select call
+      saga.next(projects);
+      // Confirm deletion
+      saga.next(true);
+
+      // Verify that it prompts to create a new project
+      expect(saga.next().value).toEqual(put(createNewProjectStart()));
+      expect(saga.next().done).toEqual(true);
+    });
+
+    it("logs an error when the project can't be deleted", () => {
+      const project = {
+        id: 'a',
+        name: 'apple',
+        path: '/path/to/apple',
+      };
+
+      const projects = [
+        project,
+        { id: 'b', name: 'berry', path: '/path/to/berry' },
+      ];
+
+      const saga = deleteProject({ project });
+
+      // Show dialog
+      saga.next();
+      // Confirm dialog
+      saga.next(0);
+      // Pass in the projects to the select call
+      saga.next(projects);
+
+      // Return `false` to `successfullyDeleted`
+      expect(saga.next(false).value).toEqual(
+        call(console.error, 'Project could not be deleted.')
+      );
+
+      // Ensure it bails early
+      expect(saga.next().done).toEqual(true);
+    });
+  });
+
+  describe('getNextProjectId', () => {
+    it('gets the next project', () => {
+      const projects = [{ id: 'a' }, { id: 'b' }, { id: 'c' }];
+      const idToDelete = 'b';
+
+      // $FlowF
```

Comment:
`// $FlowFixMe` maybe?

Reply:
Test passes when it's removed so don't think it's needed?